src/azure-cli/azure/cli/command_modules/keyvault/_params.py

```diff
@@ -87,6 +87,7 @@ class CLIJsonWebKeyOperation(str, Enum):
         c.argument('sku', arg_type=get_enum_type(SkuName, default=SkuName.standard.value))
         c.argument('no_self_perms', arg_type=get_three_state_flag(), help="Don't add permissions for the current user/service principal in the new vault.")
         c.argument('location', validator=get_default_location_from_resource_group)
+        c.argument('enable_soft_delete', arg_type=get_three_state_flag(), help='Enable vault deletion recovery for the vault, and all contained entities. If omitted, assume true as default value.')
```

Comment:
I have one concern here. As what we have mentioned yesterday, if there is no misunderstanding, we cannot disable the feature after enabling, right?

Reply:
> We cannot disable the feature after enabling, right?

No, we can't. This sentence will be included in the next version of REST spec: 

https://github.com/Azure/azure-rest-api-specs/pull/8301/files#diff-ba200dfe0b4825793d1b20fb873061d2R1125

"description": "Property to specify whether the 'soft delete' functionality is enabled for this key vault. If it's not set to any value(true or false) when creating new key vault, it will be set to true by default. Once it's been set to true value, it can NOT be reverted to false."

I am still working with keyvault team to change the phrasing so that we can directly map from the REST spec. 

> And if no --enable-soft-delete parameter disable is provided, it will enable soft delete for keyvault resource.

Yes. Soft-delete applies to data-plane objects too.